### ISSUE TITLE
Use XDG dirs spec for data/config

### ISSUE THREAD

**Issue by majutsushi** [universal-ctags/ctags#89]
It would be great if the configuration and other data used by fishman-ctags used the [XDG base dir spec](http://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html) instead of putting everything into the home directory. This would keep it in line with other more modern software and would prevent polluting the home directory.

**Comment by cweagans** [comment]
Sorry for the noise. Just switching the "Release blocker" tag for the "Initial release" milestone so that everything is tracked in one place.

**Comment by masatake** [comment]
xcmd is gone. Therefore this issue is much simplified.
Both Exuberant-ctags and Universal-ctags accesses ~/.ctags may be a trouble because Universal-ctags introduces many options and incompatible changes in command line handling.

One of the candidate is ~/.u-ctags.d/ directory or XDG base dir spec mentioned here.

**Comment by masatake** [comment]
I would like to remove /etc/ctags( and ctagd.d). It is also shared with the both implementations.
I wonder what I should do about windows.

**Comment by masatake** [comment]
Temporary I decide not introducing XDG dir.

**Comment by blueyed** [comment]
It would have been nice to go with the XDG spec when moving config to `~/.ctags.d` anyway.
I would still think that using `$XDG_CONFIG_HOME` (defaulting to `~/.config`) as a prefix makes sense, i.e. `~/.config/ctags.d` then by default?!

**Comment by justinmk** [comment]
Very much agreed with @blueyed . After reading the long discussion at #1519 I'm completely confused about why a new directory was chosen, it doesn't follow XDG conventions, and the same path can't be used for Windows...?

BTW creating a dot-prefixed directory in Windows is not difficult. Anyone using ctags knows how to use the shell to do so; why would they use Explorer...?

**Comment by cweagans** [comment]
If `~/.config/ctags.d` is what we'd need to do in order to follow XDG, I guess that's okay, but my preference would be to drop the `.d` (i.e. just `~/.config/ctags`)

**Comment by fishman** [comment]
@cweagans  same here. If XDG is a given there's no point in keeping the .d

**Comment by dsifford** [comment]
@masatake @cweagans Should this issue be reopened? Has support ever been added for this/is this still being considered?

**Comment by masatake** [comment]
@dsifford, you can reopen it, of  coruse. However, I will not work on the item till releasing.
You can make a pull request to implement what you want. As far as you update the documents in the pull request, I will review.

I don't want to take my time for impriving ctags more for this item.

**Comment by cweagans** [comment]
Reopening in case someone wants to tackle it.

**Comment by masatake** [comment]
I merged @itchyny's patch. See #2386. However, I wrapped the changes with #ifndef WIN32/#endif.

**Comment by BachoSeven** [comment]
@masatake did #2386 solve this?

**Comment by masatake** [comment]
It seems that the answer is yes.
```
[yamato@dev64]~% strace -e openat ctags
openat(AT_FDCWD, "/etc/ld.so.cache", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libxml2.so.2", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libjansson.so.4", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libyaml-0.so.2", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libseccomp.so.2", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libc.so.6", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libz.so.1", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/liblzma.so.5", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/lib64/libm.so.6", O_RDONLY|O_CLOEXEC) = 3
openat(AT_FDCWD, "/home/yamato/.config/ctags", O_RDONLY|O_NONBLOCK|O_CLOEXEC|O_DIRECTORY) = -1 ENOENT (No such file or directory)
...
```

**Comment by hholst80** [comment]
> Both Exuberant-ctags and Universal-ctags accesses ~/.ctags may be a trouble because Universal-ctags introduces many options and incompatible changes in command line handling.

This is fine.  XDG is mostly a cargo cult and often misunderstood by developers in how it should be implemented.  If you have just a simple config I'd say look at both `$XDG_CONFIG_HOME`, then it's default `~/.config` and finally under `$HOME/.ctags`.

`.ctags.d` looks horrific it is the worst combination of XDG and the `legacy` method.

**Comment by masatake** [comment]
As we merged #3972, the issue is solved.
